### PR TITLE
[23.1] Fix allowlist deserialization in file sources

### PR DESCRIPTION
**Pull request by mvdbeek**
Should fix the remaining tests

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/config/parsers.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+from galaxy.util import unicodify
+
+IpAddressT = Union[ipaddress.IPv4Address, ipaddress.IPv6Address]
+IpNetworkT = Union[ipaddress.IPv4Network, ipaddress.IPv6Network]
+IpAllowedListEntryT = Union[IpAddressT, IpNetworkT]
+
+
+def parse_allowlist_ips(fetch_url_allowlist: List[str]) -> List[IpAllowedListEntryT]:
```
Comment (jmchilton) [inline lib/galaxy/config/parsers.py]: I would put this in galaxy-util to really simplify the packages that are needed for galaxy-files. This dependency structure seems heavy handed to me.

Comment (mvdbeek) [reply lib/galaxy/config/parsers.py]: It does belong here though, doesn't it ? And file sources need config values, so this doesn't seem so outlandish that you'd bring in galaxy-config ?
The extra packages this brings in are
```
    boltons
    jinja2
    pykwalify
    PyYAML
```
afaict

Comment (mvdbeek) [reply lib/galaxy/config/parsers.py]: i've made the change but I think we will bring in more stuff that doesn't belong in util in the future.

**Review by martenson — APPROVED** [review]

File: lib/galaxy/config/parsers.py (hunk 1 above)
Comment (jmchilton) [reply lib/galaxy/config/parsers.py]: I don’t think it does belong there for it is worth. I think of the galaxy-config package as the huge app config loading process. I’d love for individual packages and components to have much simpler protocols for configuration that are isolated and don’t require all that heavy infrastructure and complex environment logic. I could totally be wrong though - maybe config is smaller than I thought or these packages are needed in util. I’ll happily approve the PR in either form.

Comment (mvdbeek) [reply lib/galaxy/config/parsers.py]: I guess we'll see how that pans out, package-specific config sounds great ... although this PR is probably a good example of how that'll be tricky 😆

**Pull request merged**